### PR TITLE
GAIA: Update default Gaia data release to DR3

### PR DESCRIPTION
**Pull request by AlinderS**
I updated the default data release for Gaia from DR2, released in April 2018, to DR3, released in June 2022.

### CODE REVIEW

**Review by ceb8 — APPROVED** [review]
LGTM

**Comment by bsipocz** [comment]
I would ping @jespinosaar  and/or @mhsarmiento to sign off on this before merging it.

File: docs/gaia/gaia.rst [hunk 1]
```diff
@@ -75,11 +75,11 @@ Examples
 
 This query searches for all the objects contained in an arbitrary rectangular projection of the sky.
 
-It is possible to choose which data release to query, by default the Gaia DR2 catalogue is used. For example::
+It is possible to choose which data release to query, by default the Gaia DR3 catalogue is used. For example::
 
   >>> from astroquery.gaia import Gaia
-  >>> Gaia.MAIN_GAIA_TABLE = "gaiaedr3.gaia_source" # Select early Data Release 3
-  >>> Gaia.MAIN_GAIA_TABLE = "gaiadr2.gaia_source"  # Reselect Data Release 2, default
+  >>> Gaia.MAIN_GAIA_TABLE = "gaiadr2.gaia_source"  # Select early Data Release 2
```
Comment (eerovaher) [inline docs/gaia/gaia.rst:81]:
```suggestion
  >>> Gaia.MAIN_GAIA_TABLE = "gaiadr2.gaia_source"  # Select Data Release 2
```

**Review by bsipocz — CHANGES_REQUESTED** [review]
I see failing remote doctests, those should be fixed prior merging.

(basically, locally a `python setup.py test -P gaia -R` should pass, too, even if CI is all green)

**Comment by AlinderS** [comment]
I'll admit I don't know why that test failed. The text that was fine was copied in the same way as the text that failed. 
Regardless, it's fixed now. Thanks for the assistance!

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Rebased to resolve the recent docs changes. And given no objections were received I'm going ahead and merge this now.

Thanks @AlinderS!

**Pull request merged**